src/librustc/hir/lowering.rs

```diff
@@ -1833,8 +1835,12 @@ impl<'a> LoweringContext<'a> {
                 ExprKind::Try(ref sub_expr) => {
                     // to:
                     //
+                    // #[allow(unreachable_patterns)]
```

Comment:
Hmm, so, this `#[allow(unreachable_patterns)]` annotation presumably affects the `<expr>`, right? That seems unfortunate.  i.e., if I do `(match { ... })?`, do we still see warnings from the inner `match`?